### PR TITLE
fix zipkinv2 translation error tag handling

### PR DESCRIPTION
**Pull request by ericmustin**
**Description:** 

This PR addresses https://github.com/open-telemetry/opentelemetry-collector/issues/2214 and, by extension, https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1644. ZipkinV2 internal translation was not taking into account the `error` tag. As mentioned in this Specification PR https://github.com/open-telemetry/opentelemetry-specification/pull/1257, and confirmed by @owais , any presence of the [`error` tag should be used to indicate a span status of `Error`](https://github.com/open-telemetry/opentelemetry-specification/pull/1257/files#diff-c50b2cac17490acd52955c504c267e15e9b8f1bfd51f15e02369cb1c83071e13R149) and is also [overriding on any other tag that denotes status for zipkin](https://github.com/open-telemetry/opentelemetry-specification/pull/1257/files#diff-60d9dfa59a8a1a0776c260526b6148549d4087deb626d42732fdd7616ef7a636R56)

**Link to tracking Issue:** <Issue number if applicable>

https://github.com/open-telemetry/opentelemetry-collector/issues/2214

**Testing:**

Added a unit test to current zipkin translation tests

### CODE REVIEW

**Comment by ericmustin** [comment]
hm, oddly seeing this error

```
go test ./... in ./exporter/alibabacloudlogserviceexporter
# github.com/open-telemetry/opentelemetry-collector-contrib/exporter/alibabacloudlogserviceexporter [github.com/open-telemetry/opentelemetry-collector-contrib/exporter/alibabacloudlogserviceexporter.test]
./tracedata_to_logservice.go:143:18: span.Status().IsNil undefined (type pdata.SpanStatus has no field or method IsNil)
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/alibabacloudlogserviceexporter [build failed]
FAIL
```
Is that expected? I know some work around `isNil` has recently been done

File: translator/trace/zipkin/zipkinv2_to_traces.go [hunk 1]
```diff
@@ -164,6 +164,11 @@ func populateSpanStatus(tags map[string]string, status pdata.SpanStatus) {
 			delete(tags, tracetranslator.TagStatusMsg)
 		}
 	}
+
+	if _, ok := tags[tracetranslator.TagError]; ok {
```
Comment (bogdandrutu) [inline translator/trace/zipkin/zipkinv2_to_traces.go:168]: does the value matter? `true` or `false`?

**Comment by bogdandrutu** [comment]
Ignore the issue in the contrib tests, I sent https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1757 to fix contrib

File: translator/trace/zipkin/zipkinv2_to_traces.go (hunk 1 above)
Comment (ericmustin) [reply translator/trace/zipkin/zipkinv2_to_traces.go:168]: hmm, It looks like only lower case `true` is accepted, so i will update

ref:
- https://github.com/open-telemetry/opentelemetry-specification/pull/1257/files#diff-60d9dfa59a8a1a0776c260526b6148549d4087deb626d42732fdd7616ef7a636R56

- https://github.com/open-telemetry/opentelemetry-specification/pull/1257/files#diff-c50b2cac17490acd52955c504c267e15e9b8f1bfd51f15e02369cb1c83071e13R130

Comment (ericmustin) [reply translator/trace/zipkin/zipkinv2_to_traces.go]: Updated

**Review by MrAlias — APPROVED** [review]

**Comment by ericmustin** [comment]
@bogdandrutu failing test looks unrelated to this PR fwiw

**Comment by ericmustin** [comment]
@bogdandrutu would've been nice to see this get iincluded in v0.17 ...anything i can help with here?

**Comment by tigrannajaryan** [comment]
@ericmustin v0.17 was released yesterday. @bogdandrutu OK to merge this?

**Review by carlosalberto — APPROVED** [review]

**Comment by ericmustin** [comment]
@bogdandrutu lmk if anything else needed here to merge, i believe this is relevant to some end users onboarding with otel so would love to be able to fit into next release 🙇

**Comment by ericmustin** [comment]
@bogdandrutu it's been 9 days so a bit concerned this may fall thru the cracks, just a friendly ping here, this is blocking an end user so would love to be able to see this get in for the next release to help with adoption.

**Comment by carlosalberto** [comment]
@ericmustin Most of the OTel community is on break, so I would expect a review from Monday, January 4th (around that date people will start to come back). That also means no releases during the next days.

Other than that, let's definitely get an early review by then. cc @tigrannajaryan

File: translator/trace/zipkin/zipkinv2_to_traces_test.go [hunk 2]
```diff
@@ -128,3 +143,23 @@ func generateTraceSingleSpanMinmalResource() pdata.Traces {
 	rsc.Attributes().UpsertString(conventions.AttributeServiceName, "SoleAttr")
 	return td
 }
+
+func generateTraceSingleSpanErrorStatus() pdata.Traces {
```
Comment (pmm-sumo) [inline translator/trace/zipkin/zipkinv2_to_traces_test.go:147]: It's a nit but perhaps [`generateTraceOneSpanOneTraceID`](https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/trace/zipkin/traces_to_zipkinv2_test.go#L109) (or just `testdata.GenerateTraceDataOneSpan`) could be reused to make this even simpler?

**Review by pmm-sumo — APPROVED** [review]

**Review by owais — APPROVED** [review]

**Review by tigrannajaryan — APPROVED** [review]
LGTM, just waiting for @bogdandrutu to make sure he does not have any further comments.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**